packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -276,168 +279,166 @@ export const initProvider = () => (dispatch: Dispatch) => {
     return decision.promise;
 };
 
-export const fetchMetadata =
-    (deviceStateArg?: string) => async (dispatch: Dispatch, getState: GetState) => {
-        const provider = selectSelectedProviderForLabels(getState());
-        if (!provider) return;
+const setMetadata =
+    ({
+        provider,
+        fileName,
+        data,
+    }: {
+        provider: MetadataProvider;
+        fileName: string;
+        data: WalletLabels | AccountLabels | undefined;
+    }) =>
+    (dispatch: Dispatch) => {
+        dispatch({
+            type: METADATA.SET_DATA,
+            payload: {
+                provider,
+                data: {
+                    [fileName]: JSON.parse(JSON.stringify(data)),
```

Comment:
why it this (stringify&parse) needed?

Reply:
should not be needed, remainder after some debugging. thanks d08c54b652e7f89f441c7bfdc6a77dd481bda0a0